ui/desktop/src/components/BaseChat.tsx

```diff
@@ -200,16 +201,25 @@ function BaseChatContent({
   // Reset recipe usage tracking when recipe changes
   useEffect(() => {
     if (recipeConfig?.title !== currentRecipeTitle) {
+      const previousTitle = currentRecipeTitle;
       setCurrentRecipeTitle(recipeConfig?.title || null);
-      setHasStartedUsingRecipe(false);
 
-      // Clear existing messages when a new recipe is loaded
-      if (recipeConfig?.title && recipeConfig.title !== currentRecipeTitle) {
+      // Only reset usage if we're switching between different recipes
+      // Don't reset if we're going from no recipe to a recipe (initial load)
+      // or if we already have messages (ongoing conversation)
```

Comment:
I think we have this comment because the code is hard to follow; let's see if we can fix that and drop the comment instead